### PR TITLE
ADEN-12861 Refactor of TVGuide slots logic based on events

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: docs/adr/ad-unit-for-layout-initializer-request.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Ad unit for LIS (Layout Initializer requeSt)
```
Comment (nandy-andy) [inline docs/adr/ad-unit-for-layout-initializer-request.md:1]: The uppercase letters were used to demonstrate where LIS acronym is coming from ;p

File: src/communication/event-types.ts [hunk 2]
```diff
@@ -251,6 +251,19 @@ export const eventsRepository: Dictionary<EventOptions> = {
 		category: '[Platform]',
 		name: 'Before page change',
 	},
+	PLATFORM_PAGE_CHANGED: {
+		category: '[Platform]',
+		name: 'Page changed',
+	},
+	PLATFORM_PAGE_EXTENDED: {
```
Comment (nandy-andy) [inline src/communication/event-types.ts:258]: Is it really needed? I thought the idea was to reuse `PLATFORM_AD_PLACEMENT_READY` but maybe in reality it wouldn't work 🤔

File: src/platforms/shared/tracking/data-warehouse.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ export class DataWarehouseTracker {
 			lc: context.get('wiki.wgUserLanguage') || 'unknown',
 			s: targetingService.get('skin') || 'unknown',
 			ua: window.navigator.userAgent,
-			u: trackingOptIn.isOptedIn() ? context.get('userId') || 0 : -1,
+			u: trackingOptIn.isOptedIn() ? context.get('options.userId') || 0 : -1,
```
Comment (nandy-andy) [inline src/platforms/shared/tracking/data-warehouse.ts:52]: Shouldn't we extract it to a separate branch and PR as it's not really connected to TV Guide events and may change data in DW? 🤔

File: src/platforms/shared/utils/insert-slots.ts [hunk 4]
```diff
@@ -21,7 +21,10 @@ export function insertSlots(slotsToInsert: SlotSetupDefinition[]): void {
 				}
 			} catch (e) {
 				utils.logger(logGroup, e.message);
-				slotsContext.setState(slotCreatorConfig.slotName, false);
+
+				if (slotCreatorConfig?.slotName) {
+					slotsContext.setState(slotCreatorConfig.slotName, false);
+				}
```
Comment (nandy-andy) [inline src/platforms/shared/utils/insert-slots.ts:27]: Shouldn't we log a warning something unexpected happened? First, the `try` code throws an exception and then we don't get the `slotName` 🤔

File: src/platforms/shared/tracking/data-warehouse.ts (hunk 3 above)
Comment (nandy-andy) [reply src/platforms/shared/tracking/data-warehouse.ts:52]: Hm, nevermind, you've changed it in [ucp-desktop-base-context.setup.ts](https://github.com/Wikia/ad-engine/pull/1814/files#diff-24ff37b813ed8d698bf699466825ee1032fae4391a1d48fd19b6889db5054298), so it's the same data.

File: docs/adr/ad-unit-for-layout-initializer-request.md (hunk 1 above)
Comment (dmnsobczak) [reply docs/adr/ad-unit-for-layout-initializer-request.md:1]: Ok, I'll do it :)

File: spec/ad-bidders/prebid/identity-hub/identity-hub.spec.ts [hunk 5]
```diff
@@ -11,7 +11,7 @@ describe('Pubmatic IdentityHub', () => {
 		loadScriptSpy = global.sandbox.stub(utils.scriptLoader, 'loadScript');
 		loadScriptSpy.resolvesThis();
 		contextStub = global.sandbox.stub(context);
-		contextStub.get.withArgs('pubmatic.identityHub.enabled').returns(true);
```
[the remark this answers is not in the document]
Comment (dmnsobczak) [reply spec/ad-bidders/prebid/identity-hub/identity-hub.spec.ts:14]: This was cherry-picked from here: https://github.com/Wikia/ad-engine/pull/1779
Simple refactor and I did it to reduce the number of conflicts and merge two improvements. UAP's PR will be merged first and those line will disappear from log :D

File: src/platforms/shared/tracking/data-warehouse.ts (hunk 3 above)
Comment (dmnsobczak) [reply src/platforms/shared/tracking/data-warehouse.ts:52]: Yep, this is also a cherry-picked change.

File: src/communication/event-types.ts (hunk 2 above)
Comment (dmnsobczak) [reply src/communication/event-types.ts:258]: Yes, but we need an event that page targeting changed. This events needs to be tested and then we decide if it's ok, what needs to be changed/fixed or if we can merge it with `PAGE_CHANGED` event.

File: src/platforms/news-and-ratings/tvguide/tvguide-platform.ts [hunk 6]
```diff
@@ -58,51 +54,103 @@ export class TvGuidePlatform {
...
-						NewsAndRatingsBaseContextSetup,
-						TvGuideTargetingSetup,
-						NewsAndRatingsTargetingSetup,
-						TvGuideSlotsContextSetup,
-						TvGuidePageChangeAdsObserver,
-						TvGuideSeamlessContentObserverSetup,
-					)
-					.execute();
-			}
+	setupSinglePageAppWatchers(container: Container) {
```
Comment (szymon-nawrocki) [inline src/platforms/news-and-ratings/tvguide/tvguide-platform.ts:117]: Can we move this to the class and run it in the main pipeline?

**Review by kwfandom — APPROVED** [review]

**Pull request merged**